vyper/parser/stmt.py

```diff
@@ -322,7 +322,11 @@ def call(self):
             return external_call.make_external_call(self.stmt, self.context)
 
     def parse_assert(self):
+        tmp = self.context.is_constant # backup value
+        self.context.is_constant = 'an assert'
```

Comment:
I believe it makes much more sense to have a separate flag/structure for 'in asssert expression' instead, see blockscope/in_assignment/in_for_loop -  using is_constant as a string is a bit hackish.